packages/edit-site/src/components/navigation-sidebar/navigation-panel/new-template-dropdown.js

```diff
@@ -20,15 +20,27 @@ import { Icon, plus } from '@wordpress/icons';
  * Internal dependencies
  */
 import getClosestAvailableTemplate from '../../../utils/get-closest-available-template';
-import { TEMPLATES_DEFAULT_DETAILS } from '../../../utils/get-template-info/constants';
+import { TEMPLATES_STATUSES } from './constants';
 
 export default function NewTemplateDropdown() {
-	const templates = useSelect(
-		( select ) =>
-			select( 'core' ).getEntityRecords( 'postType', 'wp_template', {
-				status: [ 'publish', 'auto-draft' ],
-				per_page: -1,
-			} ),
+	const { defaultTemplateTypesDefinitions, templates } = useSelect(
+		( select ) => {
+			const { getDefaultTemplateTypesDefinitions } = select(
+				'core/edit-site'
+			);
+			const templateEntities = select( 'core' ).getEntityRecords(
+				'postType',
+				'wp_template',
+				{
+					status: TEMPLATES_STATUSES,
+					per_page: -1,
+				}
+			);
+			return {
+				defaultTemplateTypesDefinitions: getDefaultTemplateTypesDefinitions(),
```

Comment:
It might be enough to call it `templateTypes` rather than `defaultTemplateTypesDefinitions `. Is there any place where we have a modified version of template types?

Reply:
We might cut `definitions` out (here especially is just confusing), but I'd keep the `default`.

The list serves two purposes:
- It's a list of "known" template types. E.g. `single` is always known; `single-{ postType }` might not be relevant for all sites.
- It contains the details (title, description) of those template types. These details will be used across WP as a "fallback": if a `wp_template` with slug `index` doesn't have a title, we'll pick it up from this list.

The list is defined by Core Gutenberg and filterable by themes/plugins.

For example a theme introducing a `book` post type, could provide a `archive-book` template file, and add its details to this list (e.g. `[ 'archive-book' => [ 'title' => 'Books Archive' ] ]`.
This way, the `archive-book` template will be displayed as "Books Archive" (localized, even) instead of `archive-book`.

At that point the user can edit the template and change its title into something else, and this custom title will have the precedence over the default one picked from the list.

Reply:
See: 6d1299d